### PR TITLE
rustdoc: inline sidebar items, to display more in a page

### PR DESCRIPTION
**Pull request by liigo**
preview link: http://liigo.com/tmp/rust/inline-sidebar-items/std/cell/index.html

---

inline-sidebar-items

### CODE REVIEW

**Comment by killercup** [comment]
This will make it pretty hard to tap on the right link on touch devices.

**Comment by joelgallant** [comment]
Re: touch devices - just specify in css to use old format on small screens? I find the scrolling to find things frustrating as-is (i end up always ctrl-f'ing), but maybe it's just me.

**Comment by liigo** [comment]
The sidebar is not visible on my mobile device (Moto X 2).

stdcell

**Comment by GuillaumeGomez** [comment]
I **really** don't like this change. If the sidebar doesn't appear on mobile, I can make a fix for it if people want it.

**Comment by liigo** [comment]
@brson I remember you said the current sidebar is useless?

**Comment by frewsxcv** [comment]
I don't feel too strongly, though I have aesthetic concerns in the scenario where the module/function/struct names are long.

**Comment by alexcrichton** [comment]
I personally prefer the list we have today, but I'm also no UI expert

**Comment by nrc** [comment]
I'm not a fan of this - it makes scanning for a name more difficult and that is the primary purpose of the sidebar - quick location of something. It is also hard to parse - the brain wants to read sentences when you have this dense display. I have never been irritated by having to scroll.

I would like some changes to the sidebar, but more about the content rather than the layout.

**Comment by steveklabnik** [comment]
It seems as though most people are :-1: on this. Thank you @liigo but I don't think we can accept it.
